drivers/regulator/regulator_da1469x.c

```diff
@@ -413,3 +468,36 @@ static int regulator_da1469x_init(const struct device *dev)
 	REGULATOR_DA1469X_DEFINE_COND(inst, v30, V30)                          \
 
 DT_INST_FOREACH_STATUS_OKAY(REGULATOR_DA1469X_DEFINE_ALL)
+
+#if !IS_ENABLED(CONFIG_PM_DEVICE) && IS_ENABLED(CONFIG_PM)
+
+void regulators_da1469x_pm_entry(enum pm_state state)
+{
+}
+
+void regulators_da1469x_pm_exit(enum pm_state state)
+{
+	if (state == PM_STATE_STANDBY &&
+	    CRG_TOP->ANA_STATUS_REG & CRG_TOP_ANA_STATUS_REG_COMP_VBAT_HIGH_Msk) {
+		DCDC->DCDC_V18_REG = dcdc_state.v18;
+		DCDC->DCDC_V18P_REG = dcdc_state.v18p;
+		DCDC->DCDC_VDD_REG = dcdc_state.vdd;
+		DCDC->DCDC_V14_REG = dcdc_state.v14;
+		DCDC->DCDC_CTRL1_REG = dcdc_state.ctrl1;
+	}
+}
+
+static struct pm_notifier regulators_da1469x_pm_notifier = {
+	.state_entry = regulators_da1469x_pm_entry,
+	.state_exit = regulators_da1469x_pm_exit,
+};
+
+int regulators_da1469x_pm_init(void)
+{
+	pm_notifier_register(&regulators_da1469x_pm_notifier);
```

Comment:
this can be done by the driver itself, init function. Since the driver seems to be for a singleton device, you can avoid the usage of the generic foreach instantiation (there are a few driver in tree doing this). This way, the fact that the notifier API can't pass context (would be needed for multiinstance devices) doesn't matter.

Reply:
suggestion applied